src/package-resolver.js

```diff
@@ -447,6 +447,12 @@ export default class PackageResolver {
     return maxValidRangeManifest;
   }
 
+  getPackageRemoteResolved(manifest: Manifest): ?string {
+    const packageReference: ?PackageReference = manifest._reference;
+    const packageRemote: ?PackageRemote = packageReference ? packageReference.remote : null;
+    return packageRemote ? packageRemote.resolved : null;
```

Comment:
I don't _think_ you need to do all the null checks given the types and invariant check... Also I'm seeing that this can be more of a helper/validation logic with what you have below that could later be moved to integrity checker, i.e. `if there is a lockfile version, does the lockfile version match the resolution's resolved version?` 

```
// in resolution-map.js
export const shouldUpdateLockfile = (lockfileEntry: ?LockManifest, resolutionEntry: ?PackageReference) => {
    if (!lockfileEntry || !resolutionEntry) {
        return false;
    }

   return lockfileEntry.resolved !== resolutionEntry.remote.resolved;
}

// then call below;
if (shouldUpdateLockfile(lockManifest, resolutionManifest._reference)) {
   this.lockfile.removePattern(pattern);
}
```
maybe something like this would work? wdyt?

Reply:
sure, this looks more readable